uefi/firmwarefile.go

```diff
@@ -69,6 +95,79 @@ type FirmwareFile struct {
 	buf    []byte
 }
 
+// Validate Firmware File
+func (f *FirmwareFile) Validate() []error {
+	errs := make([]error, 0)
+	buflen := uint64(len(f.buf))
+	blankSize := [3]byte{0xFF, 0xFF, 0xFF}
+	if buflen < FileHeaderMinLength {
+		errs = append(errs, fmt.Errorf("length too small!, buffer is only %#x bytes long", buflen))
+		return errs
+	}
+
+	// Size Checks
+	fh := &f.Header
+	if fh.Size == blankSize && buflen < FileHeaderExtMinLength {
+		errs = append(errs, fmt.Errorf("file %v length too small!, buffer is only %#x bytes long for extended header",
+			fh.Name, buflen))
+		return errs
+	}
+	if fh.Size != blankSize && f.readSize() != fh.ExtendedSize {
+		errs = append(errs, fmt.Errorf("file %v size not copied into extendedsize",
+			fh.Name))
+		return errs
+	}
+	if buflen != fh.ExtendedSize {
+		errs = append(errs, fmt.Errorf("file %v size mismatch! Size is %#x, buf length is %#x",
+			fh.Name, fh.ExtendedSize, buflen))
+		return errs
+	}
+	if fh.Size == blankSize && !fh.Attributes.isLarge() {
+		errs = append(errs, fmt.Errorf("file %v using extended header, but large attribute is not set",
+			fh.Name))
+		return errs
+	}
+
+	// Header Checksums
+	var headerSize int
+	if fh.Attributes.isLarge() {
+		headerSize = FileHeaderExtMinLength
+	} else {
+		headerSize = FileHeaderMinLength
+	}
+	// Sum over header without State and IntegrityCheck.File.
+	// To do that we just sum over the whole header and subtract.
+	var sum uint8
```

Comment:
I am not familiar with how the checksum should be computed here, so I trust what you're doing, but if you have a pointer to the specs about this I appreciate

Reply:
maybe split this into a `Checksum` method?

Reply:
Done. It's now in uefi.go